pkg/oc/cli/cmd/set/deploymenthook.go

```diff
@@ -237,66 +222,84 @@ func (o *DeploymentHookOptions) Validate() error {
 		return fmt.Errorf("you must specify a command for the deployment hook")
 	}
 
-	cmdutil.WarnAboutCommaSeparation(o.Err, o.Environment, "--environment")
+	cmdutil.WarnAboutCommaSeparation(o.ErrOut, o.Environment, "--environment")
 
 	return nil
 }
 
 func (o *DeploymentHookOptions) Run() error {
-	infos := o.Infos
-	singleItemImplied := len(o.Infos) <= 1
-	if o.Builder != nil {
-		loaded, err := o.Builder.Do().IntoSingleItemImplied(&singleItemImplied).Infos()
-		if err != nil {
-			return err
+	b := o.Builder().
+		WithScheme(ocscheme.ReadingInternalScheme, ocscheme.ReadingInternalScheme.PrioritizedVersionsAllGroups()...).
+		LocalParam(o.Local).
+		ContinueOnError().
+		NamespaceParam(o.Namespace).DefaultNamespace().
+		FilenameParam(o.ExplicitNamespace, &o.FilenameOptions).
+		Flatten()
+
+	if !o.Local {
+		b = b.
+			ResourceNames("deploymentconfigs", o.Resources...).
+			LabelSelectorParam(o.Selector).
+			Latest()
+		if o.All {
+			b = b.ResourceTypes("deploymentconfigs").SelectAllParam(o.All)
 		}
-		infos = loaded
+
+	}
+
+	singleItemImplied := false
+	infos, err := b.Do().IntoSingleItemImplied(&singleItemImplied).Infos()
+	if err != nil {
+		return err
 	}
 
 	patches := CalculatePatchesExternal(infos, func(info *resource.Info) (bool, error) {
 		dc, ok := info.Object.(*appsv1.DeploymentConfig)
 		if !ok {
 			return false, nil
 		}
-		updated, err := o.updateDeploymentConfig(dc)
-		return updated, err
+		return o.updateDeploymentConfig(dc)
 	})
 
-	if singleItemImplied && len(patches) == 0 {
-		return fmt.Errorf("%s/%s is not a deployment config or does not have an applicable strategy", infos[0].Mapping.Resource, infos[0].Name)
-	}
-
-	if len(o.Output) > 0 || o.Local || kcmdutil.GetDryRunFlag(o.Cmd) {
-		return o.PrintObject(infos)
-	}
+	// if singleItemImplied && len(patches) == 0 {
+	// 	return fmt.Errorf("%s/%s is not a deployment config or does not have an applicable strategy", infos[0].Mapping.Resource, infos[0].Name)
```

Comment:
Is this leftover, or are we enabling this error message in the future?

Reply:
Yeah, just fixed.